src/Umbraco.Web/Editors/TinyMceController.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Net;
+using System.Net.Http;
+using System.Web.Http;
+using Umbraco.Core.Services;
+using Umbraco.Web.WebApi;
+using Umbraco.Core;
+using Umbraco.Web.Mvc;
+using Umbraco.Core.IO;
+using System.IO;
+using System.Threading.Tasks;
+using Umbraco.Web.Composing;
+using Umbraco.Core.Configuration.UmbracoSettings;
+using System.Linq;
+using System;
+
+namespace Umbraco.Web.Editors
+{
+    [PluginController("UmbracoApi")]
+    public class TinyMceController : UmbracoAuthorizedApiController
+    {
+        private IMediaService _mediaService;
+        private IContentTypeBaseServiceProvider _contentTypeBaseServiceProvider;
+
+        public TinyMceController(IMediaService mediaService, IContentTypeBaseServiceProvider contentTypeBaseServiceProvider)
+        {
+            _mediaService = mediaService;
+            _contentTypeBaseServiceProvider = contentTypeBaseServiceProvider;
+        }
+
+        [HttpPost]
+        public async Task<HttpResponseMessage> UploadImage()
```

Comment:
@warrenbuckley there is no authorization assigned to this controller - this means that all back office users regardless of what they have access to can upload files. We probably want to limit the access of this controller to users that have access to content, media or member trees since those are the only places where the rte will exist?